pkg/replacer/image/image.go

```diff
@@ -202,6 +229,53 @@ func GetImageDigestFromRef(ctx context.Context, imageRef, platform string, cache
 	}, nil
 }
 
-func shouldExclude(ref string) bool {
-	return ref == "scratch"
+func shouldSkipImageRef(cfg *config.Config, ref string) bool {
+	// Parse the image reference
+	nameRef, err := name.ParseReference(ref)
+	if err != nil {
+		// we wouldn't know how to resolve this reference, so let's skip
+		return true
+	}
+
+	fullRepositoryName := nameRef.Context().Name()
+	parts := strings.Split(fullRepositoryName, "/")
```

Comment:
I didn't find a better name to parse out the image name from the full repo path...

Reply:
Let's put this under a function, add a TODO item, and live with this for now. No biggie.